### ISSUE TITLE
Polytopic Input Constraints in Learning-Based Optimal Control Using Neural Networks. (arXiv:2105.03376v1 [eess.SY])

### ISSUE THREAD

**Issue by CoffeeKumazaki** [CoffeeKumazaki/arXiv#6726]
Polytopic Input Constraints in Learning-Based Optimal Control Using Neural Networks. (arXiv:2105.03376v1 [eess.SY])<br>
https://ift.tt/3exCnfR<br>
<p>This work considers artificial feed-forward neural networks as parametric approximators in optimal control of discrete-time systems. Two different approaches are introduced to take polytopic input constraints into account. The first approach determines (sub-)optimal inputs by the application of gradient methods. Closed-form expressions for the gradient of general neural networks with respect to their inputs are derived. The approach allows to consider state-dependent input constraints, as well as to ensure the satisfaction of state constraints by exploiting recursive reachable set computations. The second approach makes use of neural networks with softmax output units to map states into parameters, which determine (sub-)optimal inputs by a convex combination of the vertices of the input constraint set. The application of both approaches in model predictive control is discussed, and results obtained for a numerical example are used for illustration.</p>
<br>
<br>
via eess.SY updates on arXiv.org http://arxiv.org/<br>
